### PR TITLE
feat: add getAll function to formdata

### PR DESCRIPTION
**Pull request by matinzd**
## Summary
The getAll() method of the [FormData](https://developer.mozilla.org/en-US/docs/Web/API/FormData) interface returns all the values associated with a given key from within a FormData object.

## Changelog

[General] [Added] - Add getAll function to FormData class for getting all parts containing that key. This is also available in web API.

## Test Plan

New test added in FormData-test.js

### CODE REVIEW

File: Libraries/Network/__tests__/FormData-test.js [hunk 1]
```diff
@@ -55,4 +55,27 @@ describe('FormData', function() {
...
+      type: 'video/mp4',
+      name: 'video.mp4',
+    });
+
+    expect(formData.getAll('video').length).toBe(1);
+
+    ['image', 'video'].forEach(type => {
+      const parts = formData.getAll(type);
+      parts.forEach(part => {
+        expect(part.fieldName).toBe(type);
```
Comment (sshic) [inline Libraries/Network/__tests__/FormData-test.js:77]: The array returned by [FormData.getAll](https://l.facebook.com/l.php?u=https%3A%2F%2Fdeveloper.mozilla.org%2Fen-US%2Fdocs%2FWeb%2FAPI%2FFormData%2FgetAll&h=AT1DICyBzwY0RFcIL2ytCyiabGZwEJhE7tW--cZnqSv7SbB75QrEnnVX5Ami2RSQVFdZAc0M9I3eqah_FPaajAwA6YaRTLEw0-FDoTIk0EFyjRBCSd7fD9Dloke4hLMdeC_-FAl_Dgpx2mzn3s9v8dRsn_k) should contain the values. It should not be the internal object representation in FormData.

File: Libraries/Network/FormData.js [hunk 2]
```diff
@@ -64,27 +64,39 @@ class FormData {
...
+      if (typeof value.type === 'string') {
+        headers['content-type'] = value.type;
       }
-      // Convert non-object values to strings as per FormData.append() spec
-      return {string: String(value), headers, fieldName: name};
+      return {...value, headers, fieldName: name};
+    }
+    // Convert non-object values to strings as per FormData.append() spec
+    return {string: String(value), headers, fieldName: name};
+  }
```
Comment (sshic) [inline Libraries/Network/FormData.js:87]: This should be hoisted off the class into a private, module-scoped function.

File: Libraries/Network/__tests__/FormData-test.js (hunk 1 above)
Comment (matinzd) [reply Libraries/Network/__tests__/FormData-test.js]: Fixed under 0c33daad69a.

**Comment by matinzd** [comment]
> Also can you fix the failed checks @[matinzd](https://github.com/matinzd)?

@sshic 

Hey Lulu,
It seems that the first two tests have some problems on circle ci because they cannot build or install dependencies. I cannot fix them.

**Comment by cortinico** [comment]
> It seems that the first two tests have some problems on circle ci because they cannot build or install dependencies. I cannot fix them.

The two builds failures are unrelated and are currently broken on main. We're looking into them and can be disregarded here.

**Comment by matinzd** [comment]
@cortinico Any update?

File: Libraries/Network/__tests__/FormData-test.js [hunk 3]
```diff
@@ -55,4 +55,33 @@ describe('FormData', function () {
     };
     expect(formData.getParts()[0]).toMatchObject(expectedPart);
   });
+
+  it('should return values based on the given key', function () {
+    formData.append('username', 'Chris');
+    formData.append('username', 'Bob');
+
+    expect(formData.getAll('username').length).toBe(2);
```
Comment (lunaleaps) [inline Libraries/Network/__tests__/FormData-test.js:63]: Nit: We could add a no-match test, where we get an empty array, otherwise this looks good to me

**Review by lunaleaps — APPROVED** [review]

File: Libraries/Network/__tests__/FormData-test.js (hunk 3 above)
Comment (matinzd) [reply Libraries/Network/__tests__/FormData-test.js:72]: Added under #d176edbc681

**Comment by cortinico** [comment]
@matinzd those 2 failures were related to an Android issue that is now fixed on master. Don't you mind rebasing?
